docs/source/data/subscriptions.mdx

```diff
@@ -366,159 +393,154 @@ subscription IncrementingNumber {
-  async onConnect(connectionParams) {
-    if (connectionParams.authorization) {
-      const currentUser = await findUser(connectionParams.authorization);
-      return { currentUser };
-    }
-    throw new Error('Missing auth token!');
-  }
-});
-```
-
-This is especially important because metadata like auth tokens are sent differently depending on the transport.
-
-## `onConnect` and `onDisconnect`
+**For subscriptions,** you can extract information from a client's request by adding [options](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.ServerOptions.md) to the first argument passed to the `useServer` function.
 
-You can define functions that the subscription server executes whenever a subscription request connects (`onConnect`) or disconnects (`onDisconnect`).
+For example, you can supply a [`context` property](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.ServerOptions.md#context) to add values of your GraphQL operation [`context`](./resolvers/#the-context-argument):
 
-Defining an `onConnect` function provides the following benefits:
-
-* You can reject a particular incoming connection by throwing an exception or returning `false` in `onConnect`. This is especially useful for [authentication](#example-authentication-with-onconnect).
-* If `onConnect` returns an object, that object is passed as the _operation_ `context` to your resolvers.
-
-You provide these function definitions to the constructor object of a `SubscriptionServer`, like so:
-
-```js{5,10}
-SubscriptionServer.create({
-  schema,
-  execute,
-  subscribe,
-  onConnect(connectionParams, webSocket, context) {
-    console.log('Connected!')
-  },
-  onDisconnect(webSocket, context) {
-    console.log('Disconnected!')
+```ts
+// ...
+useServer(
+  {
+    // Our GraphQL schema.
+    schema,
+    // Adding a context property lets you add data to your GraphQL operation context
+     context: (ctx, msg, args) => {
+       // You can define your own function for setting a dynamic context
+       // or provide a static value
+      return getDynamicContext(ctx, msg, args);
+    },
   },
-});
+  wsServer,
+);
 ```
 
-These functions are passed the following parameters:
-
-<table class="field-table">
-  <thead>
-    <tr>
-      <th>Name /<br/>Type</th>
-      <th>Description</th>
-    </tr>
-  </thead>
+Now here is where things might get a little bit confusing, notice above that the `context` property's first argument is `ctx`, also known as context. _What?_ The [`Context`](https://github.com/enisdenjo/graphql-ws/blob/master/docs/interfaces/server.Context.md) type in `graphql-ws` is **not the same thing** as the GraphQL `context` argument passed to your resolvers.
 
-<tbody>
-<tr>
-<td>
+The `Context` type from `graphql-ws` is where information about your subscription server and its requests resides. You can access incoming parameters from a client `subscription` request by checking the `Context.connectionParams`.
```

Comment:
If the above suggestion looks good, this might work here:

You can access a client's WebSocket connection parameters (such as an auth token) via the `ctx.connectionParams` property.

Reply:
I think the phrase "WebSocket connection parameters" feels a bit wordy. How about a slight rephrase:

"You can access the parameters of a client's `subscription` request to your WebSocket server via the `ctx.connectionParams` property."

Reply:
Yep that looks good!